integration_tests/test_str_to_int.py

```diff
@@ -6,5 +6,18 @@ def f():
     assert i == 314
     i = int("-314")
     assert i == -314
+    s: str
+    s = "123"
+    i = int(s)
+    assert i == 123
+    s = "-123"
+    i = int(s)
+    assert i == -123
+    s = "    1234"
+    i = int(s)
+    assert i == 1234
+    s = "    -1234   "
+    i = int(s)
+    assert i == -1234
```

Comment:
Let's check for something like, `int("2030.300")`. Supposed to give an error as follows,

```bash
Python 3.8.9 (default, Apr 13 2022, 08:48:06) 
[Clang 13.1.6 (clang-1316.0.21.2.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> int("2030.300")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '2030.300'
```

Reply:
Unfortunately, this is very tricky. There are so many cases to consider. See how CPython does this:

https://github.com/python/cpython/blob/4b4439daed3992a5c5a83b86596d6e00ac3c1203/Objects/longobject.c#L2270-L2271

I suggest that we deal with this in a separate PR. Also, the changes in https://github.com/lcompilers/lpython/pull/591/files were incomplete.

Reply:
The above input gives this error as expected, right ? It is a good test to have. I'll look into CPython examples for missed cases.

semantic error: invalid literal for int() with base 10: '2030.300'
 --> try.py:5:13
   |
 |     i = int("2030.300")